### PR TITLE
Adding `annotate` module in the `cp_process_singlecells` workflow 

### PR DESCRIPTION
**Pull request by axiomcura**
## About PR

In this PR, the `cp_process_singlecells` will have the annotation module imported for analysis.

### CODE REVIEW

File: configs/wf_configs/cp_process_singlecells.yaml [hunk 1]
```diff
@@ -42,6 +48,23 @@ cytotable_convert:
...
 
+annotate_configs:
+  params:
+    join_on:
+      - Metadata_well_position
+      - Image_Metadata_Well
+    add_metadata_id_to_platemap: True
+    format_broad_cmap: False
+    clean_cellprofiler: True
+    external_metadata: "none"
```
Comment (MikeLippincott) [inline configs/wf_configs/cp_process_singlecells.yaml:59]: I don't have a lot of experience with what the norm is for these `.yaml` files but is it possible to add hints to what the possibilities are for each param? ex. `dest_datatype: parquet` what are the possible input types? I am not sure is this is doable or a norm just a thought

File: workflows/rules/annotate.smk [hunk 2]
```diff
@@ -24,19 +24,20 @@ Returns:
...
+        profile=CYTOTABLE_CONVERTED_PLATE_DATA,
         barcodes=BARCODES,
         metadata=METADATA_DIR,
     output:
         ANNOTATED_DATA,
     conda:
         "../envs/cytominer_env.yaml"
     log:
-        "logs/annotate_{file_name}.log",
+        "logs/annotate_{basename}.log",
```
Comment (MikeLippincott) [inline workflows/rules/annotate.smk:40]: less ambiguous, I like basename better!

File: workflows/workflow/cp_process_singlecells.smk [hunk 3]
```diff
@@ -34,6 +35,7 @@ configfile: "./configs/wf_configs/cp_process_singlecells.yaml"
 # importing modules
 include: "../rules/common.smk"
 include: "../rules/cytotable_convert.smk"
+include: "../rules/annotate.smk"
```
Comment (MikeLippincott) [inline workflows/workflow/cp_process_singlecells.smk:38]: Looking at line 30-32: how do multiple config files know which rules they belong to?

**Review by MikeLippincott — APPROVED** [review]
LGTM. Left a couple of comments to address. Minor documentation.

File: workflows/workflow/cp_process_singlecells.smk (hunk 3 above)
Comment (axiomcura) [reply workflows/workflow/cp_process_singlecells.smk:38]: Each module represented by the `include` takes in the same workflow config.

Here's an example from: https://github.com/WayScience/CytoSnake/blob/aa852f9f4886057cb96f5aa19e648e2d760adc22/configs/wf_configs/cp_process_singlecells.yaml

```yaml
cytotable_convert:
  params:
    dest_datatype: parquet
    source_datatype: sqlite
    concat: True
    join: True
    infer_common_schema: True
    drop_null: True
    preset: cellprofiler_sqlite
    log_level: ERROR

annotate_configs:
  params:
    join_on:
      - Metadata_well_position
      - Image_Metadata_Well
    add_metadata_id_to_platemap: True
    format_broad_cmap: False
    clean_cellprofiler: True
    external_metadata: "none"
    external_join_left: "none"
    external_join_right: "none"
    compression_options:
      method: "gzip"
      mtime: 1
    float_format: null
    cmap_args: {}
```

The corresponding parameters are selected by the name of the module. Above presents the parameters for both:

```yaml 
include: "../rules/cytotable_convert.smk"
include: "../rules/annotate.smk"
```

Here's an simple image:
 
Screenshot 2023-05-19 at 7 14 10 AM

**Comment by axiomcura** [comment]
Thanks for the discussion points, merging!

**Pull request merged**